### PR TITLE
Run bootstrap demo currency with parachain

### PR DESCRIPTION
**Pull request by clangenb**
Make client compatible with parachain

### CODE REVIEW

**Comment by brenzi** [comment]
I don't like the timeout. What about waiting for two blocks?
```
$CLIENT listen -b 2
```

Are you sure Charlie is always endowed? also for old nodes?

**Comment by clangenb** [comment]
> Are you sure Charlie is always endowed? also for old nodes?

No, he isn't, but I reverted that in 58c0be1

> I don't like the timeout. What about waiting for two blocks?
> 
> ```
> $CLIENT listen -b 2
> ```
I am not certain if it is deterministically 2 blocks, but I will check.

**Review by brenzi — APPROVED** [review]

File: client/src/main.rs [hunk 1]
```diff
@@ -179,19 +179,9 @@ fn main() {
...
-                                GenericAddress::Id(to.clone()),
-                                PREFUNDING_AMOUNT
-                            )),
-                            nonce,
-                            Era::Immortal,
-                            _api.genesis_hash,
-                            _api.genesis_hash,
-                            _api.runtime_version.spec_version,
-                            _api.runtime_version.transaction_version
+                        let xt = _api.balance_transfer(
```
Comment (brenzi) [inline client/src/main.rs]: nono, this won't work!
The faucet should be able to batch fund multiple accounts without always awaiting the next block. So we need to increase the nonce manually, as the api-client can't know the next nonce

**Review by brenzi — CHANGES_REQUESTED** [review]

File: client/src/main.rs (hunk 1 above)
Comment (clangenb) [reply client/src/main.rs]: Ah yes, that totally makes sense. It didn't work, as the `Call::Balances` enum is wrong for the parachain case, which why I made the change in the first place, but I did not realize why it was implemented like that.

In that case, I will change it back and use the `compose_call!` macro instead.

Comment (brenzi) [reply client/src/main.rs]: will this be fixed?
```                                                                                                                             
Faucet drips to 5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y (Alice's nonce=5)                                                                                                  
thread 'client' panicked at 'extrinsic error code 1010: Invalid Transaction: Transaction has a bad signature', /home/abrenzikofer/.cargo/git/checkouts/substrate-api-client-9327f54164a9792a/a6e1616/src/rpc/client.rs:173:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: RecvError', /home/abrenzikofer/.cargo/git/checkouts/substrate-api-client-9327f54164a9792a/a6e1616/src/lib.rs:454:45
```

**Comment by clangenb** [comment]
I need to investigate. I suspect this is for the rebased parachain node?
Because that did not happen before.

**Comment by brenzi** [comment]
yes, rebased parachain

File: client/bootstrap_demo_currency.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-CLIENT="../target/release/encointer-client-notee"
+CLIENT="../target/release/encointer-client-notee -p 9946"
```
Comment (brenzi) [inline client/bootstrap_demo_currency.sh]: I don't like any parachain-specific things in here. The script should run with minimal CLI options and I don't want to modify tutorials

Comment (clangenb) [reply client/bootstrap_demo_currency.sh]: I agree, which is why this was reverted in 4626f50cf112f47bcbe20d316f4bf8647dbaf876 :)

**Comment by brenzi** [comment]
tested against encointer-parachain and encointer-node-notee. works

**Pull request merged**